test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/BodyModelBinderProviderTest.cs

```diff
@@ -70,11 +71,34 @@ public void GetBinder_WhenBindingSourceIsFromBody_ReturnsBinder()
             Assert.IsType<BodyModelBinder>(result);
         }
 
+        [Fact]
+        public void ObsoleteConstructor_DoesNotThrowNullReferenceException()
+        {
+            // Arrange
+            var context = new TestModelBinderProviderContext(typeof(Person));
+            context.BindingInfo.BindingSource = BindingSource.Body;
+            var formatter = new TestInputFormatter();
+            var formatterList = new List<IInputFormatter>();
+            formatterList.Add(formatter);
+#pragma warning disable 0618
+            var provider = new BodyModelBinderProvider((formatterList), new TestHttpRequestStreamReaderFactory());
+#pragma warning restore 0618
+
+            // Act
+            var exception = Record.Exception(() => provider.GetBinder(context));
+
+            // Assert
+            Assert.Null(exception);
```

Comment:
This works but would be simpler as
``` c#
// Act & Assert (does not throw)
provider.GetBinder(context);
```
Same for similar `BodyModelBinderTests` test.